moto/iam/models.py

```diff
@@ -553,6 +553,19 @@ def create_from_cloudformation_json(
 
         return role
 
+    @classmethod
+    def delete_from_cloudformation_json(
+        cls, resource_name, cloudformation_json, region_name
+    ):
+        for profile_name, profile in iam_backend.instance_profiles.items():
+            profile.delete_role(role_name=resource_name)
```

Comment:
A minor nitpick on this implementation:
In an ideal world, the InstanceProfile#delete would read the `cloudformation_json`, get the roles that are attached to it, and delete the role only from our InstanceProfile.
Same for policies.

The current implementation could lead to a scenario where if another profile is created outside of CF, the attached role will still be deleted. AWS would throw an error in that scenario, for trying to delete a role attached to an an existing profile.